src/main/java/seedu/address/model/loan/Nric.java

```diff
@@ -0,0 +1,82 @@
+package seedu.address.model.loan;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+/**
+ * Class that stores the NRIC of a person.
+ */
+public class Nric {
+
+    public static final String MESSAGE_NRIC_CONSTRAINTS =
+            "Nric should be singapore issued. It may be blank. ";
+
+    public static final String NRIC_VALIDATION_REGEX = "^[STGF][0-9]{7}[A-Z]$";
+
+    public final String nric;
+
+    /**
+     * Constructs a {@code Nric}.
+     *
+     * @param ic A valid nric.
+     */
+    public Nric(String ic) {
+        ic = ic.trim().toUpperCase();
```

Comment:
This line should come after the next line, since it only makes sense if ic is non-null.

Also, trimming should not be done here as strings with leading and trailing spaces should be rejected. (See the test cases for the other data fields.) Trimming is already done at the command parser level with tokenize, I think.

Reply:
Good catch. Line 24 will throw a NPE if `ic` is `null`.

Reply:
Actually it doesn't make a technical difference because `requireNonNull()` specifically does nothing except to throw a NPE anyway (which will be caught and handled by the calling method so that it displays an error message to the user, blah blah.). It just matters more on a semantic level.